### PR TITLE
integrate groupbox control

### PR DESCRIPTION
**Pull request by mameolan**
## What does the pull request do?
adds the missing GroupBox Control

## Checklist

- [ ] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

File: src/Avalonia.Themes.Default/GroupBox.xaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    xmlns="https://github.com/avaloniaui"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    >
+
+  <Styles.Resources>
+    <Thickness x:Key="GroupBoxHeaderThemePadding">4</Thickness>
+  </Styles.Resources>
+  
+  <Style Selector="GroupBox">
+    <Setter Property="Background" Value="{DynamicResource ThemeAccentBrush}" />
```
Comment (danwalmsley) [inline src/Avalonia.Themes.Default/GroupBox.xaml:11]: Might need to make IsFocusable and IsTapStop false

I beleive WPF version doest this:
https://github.com/dotnet/wpf/blob/master/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/GroupBox.cs#L24

**Review by danwalmsley — CHANGES_REQUESTED** [review]
Very good, just 1 potential issue found

File: src/Avalonia.Themes.Default/GroupBox.xaml (hunk 1 above)
Comment (mameolan) [reply src/Avalonia.Themes.Default/GroupBox.xaml:11]: Ok. I will add it.

Comment (mameolan) [reply src/Avalonia.Themes.Default/GroupBox.xaml:11]: Override some metadata:
Focusable: false
TabNavigation: None

File: src/Avalonia.Controls/GroupBox.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+{
+    /// <summary>
+    /// GroupBox Control class
+    /// </summary>
+    public class GroupBox: HeaderedContentControl
+    {
+        /// <summary>
+        /// style key of this control
+        /// </summary>
+        public Type StyleKey => typeof(GroupBox);
```
Comment (Gillibald) [inline src/Avalonia.Controls/GroupBox.cs:15]: This shouldn't be needed

Comment (mameolan) [reply src/Avalonia.Controls/GroupBox.cs]: Ok. I though we have to add this line. Perhabs I misunderstood something.

Comment (danwalmsley) [reply src/Avalonia.Controls/GroupBox.cs]: only if you class is of a different type to the template...

if you class was MyCustomGroupBox, you can use this to make it get the template GroupBox has.

File: samples/ControlCatalog/MainView.xaml [hunk 3]
```diff
@@ -57,6 +57,7 @@
       <TabItem Header="TextBlock"><pages:TextBlockPage/></TabItem>
       <TabItem Header="ToolTip"><pages:ToolTipPage/></TabItem>
       <TabItem Header="TreeView"><pages:TreeViewPage/></TabItem>
+      <TabItem Header="GroupBox"><pages:GroupBoxPage/></TabItem>
```
Comment (danwalmsley) [inline samples/ControlCatalog/MainView.xaml:60]: Put this in alphabetical order.

**Review by danwalmsley — CHANGES_REQUESTED** [review]
1 last minor change

File: samples/ControlCatalog/MainView.xaml (hunk 3 above)
Comment (mameolan) [reply samples/ControlCatalog/MainView.xaml]: Ok. Done.

**Comment by danwalmsley** [comment]
@mameolan sorry about this, but any change we can have it styled like the wpf one?

image

**Comment by JamRemco** [comment]
> @mameolan sorry about this, but any change we can have it styled like the wpf one?
> 
> image

Seiously that's so win95 haha ;)

**Comment by Gillibald** [comment]
I personally dislike this control because it brings no functionality. It is just a styled HeaderedContentControl

**Comment by grokys** [comment]
For reference, this control currently looks like this:

image

> can we have it styled like the wpf one?

I don't think this would be a good idea, the WPF one is Win9x styled and we use a more UWP-style. Thing is, the `GroupBox` doesn't really _have_ a more modern style, it's a control that doesn't really exist any more. Can anyone show me an example of a `GroupBox` in a modern application?

> I personally dislike this control because it brings no functionality. It is just a styled HeaderedContentControl

Yeah, for this reason and the styling/outdatedness issues, I'm not sure we would want to accept this control into Avalonia core.

**Comment by danwalmsley** [comment]
@grokys the argument for having it could be a lot of wpf legacy code might have it, and it just makes it easier when copying and pasting examples... i remember now this is the second time someone has sent a PR for this...

im neutral on having it though.

**Comment by kekekeks** [comment]
Could we have a proper template for `HeaderedContentControl` and keep `GroupBox` as a deprecated alias?

**Comment by JamRemco** [comment]
@grokys 
 I got you point, its like template a textblock and call it a new control. however we could make a Styleclass for HeaderContentControl thats kinda looks and act like a groupbox. 
So instead of a new control you just got a style 
<HeaderedContentControl classes = "Groupnox"/>
That way Everybody's happy.

@danwalmsley  agreed people seems to like/use it .

@kekekeks how would a "proper" template looks/function in your oppinion.

**Comment by kekekeks** [comment]
@JamRemco Right now HeaderedContentControl has no template at all.

**Comment by danwalmsley** [comment]
will close for now... @mameolan sorry... hope you will continue to contribute :)

**Comment by CorruptComputer** [comment]
I actually am a fan of how these look, especially when making small UI's to command line apps. Is there no chance that this could get added still? I don't seem the harm with it, since if you don't like it you can just not use it.

**Comment by rstm-sf** [comment]
You can create a separate control, the source code is available ;)

**Comment by Shadowblitz16** [comment]
Please reopen this, I want it too.
We need more desktop controls.

**Comment by cla-avalonia** [comment]
@mameolan @danwalmsley,

Please read the following Contributor License Agreement (CLA). If you agree with the CLA, please reply with the following:
```
@cla-avalonia agree
```

<details>
    <summary>Contributor License Agreement</summary>
    
### Contribution License Agreement

This Contribution License Agreement ( **“Agreement”** ) is agreed to by the party signing below ( **“You”** ),
and conveys certain license rights to the AvaloniaUI OÜ ( **“AvaloniaUI OÜ”** ) for Your contributions to
AvaloniaUI OÜ open source projects. This Agreement is effective as of the latest signature date below.

**1. Definitions.**

**“Code”** means the computer software code, whether in human-readable or machine-executable form,
that is delivered by You to AvaloniaUI OÜ under this Agreement.

**“Project”** means any of the projects owned or managed by AvaloniaUI OÜ and offered under a license
approved by the Open Source Initiative (www.opensource.org).

**“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any
Project, including but not limited to communication on electronic mailing lists, source code control
systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
discussing and improving that Project, but excluding communication that is conspicuously marked or
otherwise designated in writing by You as “Not a Submission.”

**“Submission”** means the Code and any other copyrightable material Submitted by You, including any
associated comments and documentation.

**2. Your Submission.** You must agree to the terms of this Agreement before making a Submission to any
Project. This Agreement covers any and all Submissions that You, now or in the future (except as
described in Section 4 below), Submit to any Project.

**3. Originality of Work.**  You represent that each of Your Submissions is entirely Your 
original work. Should You wish to Submit materials that are not Your original work, 
You may Submit them separately to the Project if You (a) retain all copyright and 
license information that was in the materials as you received them, (b) in the 
description accompanying your Submission, include the phrase &quot;Submission 
containing materials of a third party:&quot; followed by the names of the third party and any 
licenses or other restrictions of which You are aware, and (c) follow any other 
instructions in the Project&#39;s written guidelines concerning Submissions.

**4. Your Employer.** References to “employer” in this Agreement include Your employer or anyone else
for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
Submission is made in the course of Your work for an employer or Your employer has intellectual
property rights in Your Submission by contract or applicable law, You must secure permission from Your
employer to make the Submission before signing this Agreement. In that case, the term “You” in this
Agreement will refer to You and the employer collectively. If You change employers in the future and
desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
and secure permission from the new employer before Submitting those Submissions.

**5. Licenses.**

**a. Copyright License.** You grant AvaloniaUI OÜ, and those who receive the Submission directly
or indirectly from AvaloniaUI OÜ, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable
license in the Submission to reproduce, prepare derivative works of, publicly display, publicly perform,
and distribute the Submission and such derivative works, and to sublicense any or all of the foregoing
rights to third parties.

**b. Patent License.** You grant AvaloniaUI OÜ, and those who receive the Submission directly or
indirectly from AvaloniaUI OÜ, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license
under Your patent claims that are necessarily infringed by the Submission or the combination of the
Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
import or otherwise dispose of the Submission alone or with the Project.

**c. Other Rights Reserved.** Each party reserves all rights not expressly granted in this Agreement.
No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
granted by implication, exhaustion, estoppel or otherwise.

**6. Representations and Warranties.** You represent that You are legally entitled to grant the above
licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
have disclosed under Section 3 ). You represent that You have secured permission from Your employer to
make the Submission in cases where Your Submission is made in the course of Your work for Your
employer or Your employer has intellectual property rights in Your Submission by contract or applicable
law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
EXPRESSLY STATED IN SECTIONS 3, 4, AND 6 , THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.

**7. Notice to AvaloniaUI OÜ.** You agree to notify AvaloniaUI OÜ in writing of any facts or
circumstances of which You later become aware that would make Your representations in this
Agreement inaccurate in any respect.

**8. Information about Submissions.** You agree that contributions to Projects and information about
contributions may be maintained indefinitely and disclosed publicly, including Your name and other
information that You submit with Your Submission.

**9. Governing Law/Jurisdiction.** This Agreement is governed by the laws of the Republic of Estonia, and
the parties consent to exclusive jurisdiction and venue in the courts sitting in Talinn,
Estonia. The parties waive all defenses of lack of personal jurisdiction and forum non-conveniens.

**10. Entire Agreement/Assignment.** This Agreement is the entire agreement between the parties, and
supersedes any and all prior agreements, understandings or communications, written or oral, between
the parties relating to the subject matter hereof. This Agreement may be assigned by AvaloniaUI OÜ.


*AvaloniaUI OÜ dedicates this Contribution License Agreement to the public domain according to the Creative Commons CC0 1.*

</details>

**0** out of **2** committers have signed the CLA.
- [ ] @mameolan
- [ ] @danwalmsley




**mameolan** doesn't seem to be a GitHub user.
Please add the emails used in your commits to your Github account accordingly.

**Pull request closed without merging**